src/coin_test/data/dataset_generator.py

```diff
@@ -100,58 +177,66 @@ def select_data(
             df_norm: Normalized Dataframe of original data
             starting_price: The first open price for the data
             num_rows: The number of rows in the dataset
+            chunk_size: The amount of rows to combine in each chunk
             rng: A numpy random number generator
 
         Returns:
             pd.DataFrame: The synthetic dataset
         """
-        # Select data
-        sampled_data = df_norm.sample(
-            n=num_rows, replace=True, random_state=rng
-        ).reset_index(drop=True)
-        sampled_data.loc[0, "Open"] = starting_price
+        # Select Chunk Length Windows
+        chunks = tuple(df_norm.rolling(chunk_size))[chunk_size - 1 :]
+        print("\n\n".join(repr(x) for x in chunks))
+        rows_per_chunk = len(chunks[0])
+        num_chunks = ceil(num_rows / rows_per_chunk)
 
-        # Stack Open and Close data in one column
-        oc_stacked = sampled_data.melt(
-            id_vars=["High", "Low"],
-            value_vars=["Open", "Close"],
-            var_name="O/C",
-            value_name="Perc_Change",
-            ignore_index=False,
-        ).reset_index()
-        oc_stacked.sort_values(
-            by=["index", "O/C"], ascending=[True, False], inplace=True
-        )
+        # Select data and concatenate chunks
+        r = Random(rng.bytes(16))  # type: ignore
```

Comment:
Is this using the existing random seed stuff?

Reply:
we used to use numpy's random number generator. however, the python builtin random object doesn't use numpy's rng object so I send bytes from this generator as the seed for the random object here (which uses random.sample)